tests/rptest/scale_tests/many_partitions_test.py

```diff
@@ -741,21 +753,26 @@ def _write_and_random_read(self, scale: ScaleParameters, topic_names):
         self.logger.info(
             "Write+randread stress test complete, verifying sequentially")
 
+        # When tiered storage is enabled, don't consume the entire topic, as
+        # that could entail millions of segments from the cloud.
+        max_msgs = None
+        if scale.tiered_storage_enabled:
+            max_msgs = 10000
```

Comment:
This is only 320MiB of data, a few seconds of traffic.  We don't have to read the whole topic, but let's read more than this.